patches/api/0359-Expand-PlayerItemMendEvent.patch

```diff
@@ -0,0 +1,58 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Thu, 20 Jan 2022 18:11:44 -0800
+Subject: [PATCH] Expand PlayerItemMendEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/player/PlayerItemMendEvent.java b/src/main/java/org/bukkit/event/player/PlayerItemMendEvent.java
+index 2f0cbf9287c954e37cc76b88477b7e8ba7a759e3..e21936ee51106a19d93d7e83250c722975f6cd00 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerItemMendEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerItemMendEvent.java
+@@ -21,13 +21,47 @@ public class PlayerItemMendEvent extends PlayerEvent implements Cancellable {
+     private final ExperienceOrb experienceOrb;
+     private int repairAmount;
+     private boolean cancelled;
++    private java.util.function.IntUnaryOperator durabilityToXpOp; // Paper
+ 
++    @Deprecated // Paper
+     public PlayerItemMendEvent(@NotNull Player who, @NotNull ItemStack item, @NotNull ExperienceOrb experienceOrb, int repairAmount) {
++        // Paper start
++        this(who, item, experienceOrb, repairAmount, amount -> amount / 2);
++    }
++
++    public PlayerItemMendEvent(@NotNull Player who, @NotNull ItemStack item, @NotNull ExperienceOrb experienceOrb, int repairAmount, java.util.function.IntUnaryOperator durabilityToXpOp) {
```

Comment:
Needs a nullability annotation on the UnaryOperator (which is why the tests are failing)